nix/devcontainer/devcontainer.nix

```diff
@@ -74,27 +83,31 @@ let
       # make sure /tmp exists
       mkdir -m 1777 tmp
 
-      # need a HOME
-      mkdir -vp root
-
       # allow ubuntu ELF binaries to run. VSCode copies it's own.
       chmod +w lib64
       ln -s ${glibc}/lib64/ld-linux-x86-64.so.2 lib64/ld-linux-x86-64.so.2
       ln -s ${gcc-unwrapped.lib}/lib64/libstdc++.so.6 lib64/libstdc++.so.6
       chmod -w lib64
-
     '' + extraCommands;
 
+    runAsRoot = ''
+      #!${runtimeShell}
+      ${dockerTools.shadowSetup}
+      groupadd --gid ${nonRootUserId} ${nonRootUser}
+      useradd --uid ${nonRootUserId} --gid ${nonRootUserId} -m ${nonRootUser}
```

Comment:
Hm, we actually have checked in `shadow` and `group` files for some reason (I really don't know why, I guess to avoid having to do the `nixbld` user setup or something...). maybe we could just add them to that?

Reply:
@michaelpj yeah it's a bit mysterious to me too.

I did think about just adding this user there; but I didn't want to do that because that doesn't establish the home directory with the right bashrc; it just seemed nicer to keep it all encapsulated in one spot.

The converse; removing those files and doing _all_ our groupadds and such here also seemed like a bad idea ...

:shrug: wasn't sure which way to go.

Reply:
Okay, this seems fine then. This whole setup is a bit of a mess, but none of us quite know how to do it better, so :shrug:

Reply:
@michaelpj i can have a look if i can delete some of that shadow stuff in any case; because i realised it's mostly nix builder users; which i suppose we don't even need in the devcontainer image anyway, given that it's not using nix?